hs_explore/migrations/0002_auto_20200625_1910.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.28 on 2020-06-25 19:10
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def fill_csdms_names(apps, schema_editor):
+    """ Store CSDMS names into LDAWord model
+    """
+    LDAWord = apps.get_model('hs_explore', 'LDAWord')
+    CSDMSName = apps.get_model('hs_csdms', 'CSDMSName')
+    csdms_names = set()
+    splitted_names = set()
+    for csdms_record in CSDMSName.objects.filter(part='name'):
+        csdms_name = csdms_record.value
+        csdms_names.add(csdms_name)
+        if len(csdms_name) <= 1:
+            continue
+        lda_word_record = LDAWord()
+        lda_word_record.source = 'CSDMS'
+        lda_word_record.word_type = 'keep'
+        lda_word_record.part = 'name'
+        lda_word_record.value = csdms_name
+        lda_word_record.save()
+        tokens = csdms_name.split(" ")
+        splitted_names.update(tokens)
+
+    for splitted_name in splitted_names:
+        if len(splitted_name) <= 1 or splitted_name in csdms_names:
+            continue
+        lda_word_record = LDAWord()
+        lda_word_record.source = 'CSDMS'
+        lda_word_record.word_type = 'keep'
+        lda_word_record.part = 'name'
+        lda_word_record.value = splitted_name
+        lda_word_record.save()
+
+
+def fill_odm2(apps, schema_editor):
```

Comment:
This looks like there is an LDAWord for every ODM2Variable.  ODM2Variable isn't static and can be updated on demand.
https://github.com/hydroshare/hydroshare/blob/develop/hs_odm2/models.py#L30

@alvacouch should chime in on this, but perhaps setup a signal in sync and tie into it.  Do not call your code directly, it's in a different app and it should not be dependent on hs_explore